Marlin/src/core/serial.h

```diff
@@ -75,9 +75,7 @@ enum DebugFlags {
     #include "../HAL/HAL_AVR/MarlinSerial.h"
```

Comment:
@thinkyhead somehow this `#ifdef __AVR__` section survived your rebasing. This section was moved into _HAL_AVR.h_ to address the todo comment, and needs to be removed from this file.

Reply:
Ah, I was dubious about that. Still needs a patch? I'll do that now.

Reply:
What should I be left with after the edit? Is this too much…?
```cpp
#ifdef __AVR__
  #ifdef USBCON
    #include <HardwareSerial.h>
  #else
    #include "../HAL/HAL_AVR/MarlinSerial.h"
  #endif
#elif defined(ARDUINO_ARCH_SAM)
  // To pull the Serial port definitions and overrides
  #include "../HAL/HAL_DUE/MarlinSerial_Due.h"
#endif
```

Reply:
Shouldn't need any of the `__AVR__` section as it's in the HAL_AVR.h already. Ideally `ARDUINO_ARCH_SAM` gets moved into the appropriate HAL.h. 

Also, checkout #9141 for another oopsie I found.

Reply:
Merged, but simply removed our custom definition of `BYTE`, which is nonstandard for `serial.print`.

Reply:
Works for me